src/main/webapp/app/entities/result/updating-result.component.ts

```diff
@@ -90,30 +80,24 @@ export class UpdatingResultComponent implements OnInit, OnChanges, OnDestroy {
             // only subscribe for the currently logged in user or if the participation is a template/solution participation and the student is at least instructor
             const isInstructorInCourse = this.participation.student == null && exercise.course && this.accountService.isAtLeastInstructorInCourse(exercise.course);
             const isSameUser = this.participation.student && user.id === this.participation.student.id;
-            const exerciseNotOver = exercise.dueDate == null || (moment(exercise.dueDate).isValid() && moment(exercise.dueDate).isAfter(moment()));
 
-            if ((isSameUser && exerciseNotOver) || isInstructorInCourse) {
```

Comment:
I removed the check for exerciseNotOver - we should still establish the websocket connection after the due date passes, as it is an option to show ungraded results in the result component.